### DISCUSSION TITLE
Disable git lfs on all branches except master

### DISCUSSION THREAD

**Discussion by nsChrys** [git-lfs/git-lfs#5483] [Q&A]
I need to track large binary files. Since each LFS object is stored as a separate file, I'm concerned that too many commits will fill up the available space very quickly. Hence, my question: Is there a way to disable git lfs on all branches except the master branch?

**Comment by bk2204** [comment]
Hey,

This isn't possible with the way Git and Git LFS works.  You certainly can have a main branch that has a `.gitattributes` pattern (which controls whether a file is an LFS file) which isn't in other branches, but as soon as you branch off the main branch again, you're going to include it in the new branch.  In addition, a merge doesn't re-clean the files, so you'll end up with files that are labelled as LFS but stored as Git files, which the FAQ explains will cause `git status` to constantly show files as changed.

However, note that objects you create and push to the remote can be removed, so if you create a branch and push it, then you can run `git lfs prune -f` to remove those objects which you no longer need on your local system.  The goal of Git LFS is to let you work with larger files by removing files you don't need from the local system and instead hosting them on the remote, so `git lfs prune` has several options which control that behaviour.  I'll refer you to the documentation for that subcommand and the `git-lfs-config` documentation for its configuration.
